influxdb_iox/src/commands/debug/wal/inspect.rs

```diff
@@ -0,0 +1,146 @@
+//! A module providing a CLI command to inspect the contents of a WAL file.
+use std::{io::Write, ops::RangeInclusive, path::PathBuf};
+
+use itertools::Itertools;
+use wal::SequencedWalOp;
+
+use super::Error;
+
+#[derive(Debug, clap::Parser)]
+pub struct Config {
+    /// The path to the input WAL file
+    #[clap(value_parser)]
+    input: PathBuf,
+
+    /// An optional range of sequence numbers to restrict the inspection to, in
+    /// the format "%d-%d". Only entries that have a sequence number falling
+    /// within the range (inclusive) will be displayed
+    #[clap(long, short, value_parser = parse_sequence_number_range)]
+    sequence_number_range: Option<RangeInclusive<u64>>,
+}
+
+fn parse_sequence_number_range(s: &str) -> Result<RangeInclusive<u64>, String> {
+    let parts: Vec<&str> = s.split('-').collect();
+    if parts.len() != 2 {
+        return Err("sequence number range provided does not use format <START>-<END>".to_string());
+    }
+
+    let min = parts[0]
+        .parse()
+        .map_err(|_| format!("{} isn't a valid sequence number", parts[0]))?;
+    let max = parts[1]
+        .parse()
+        .map_err(|_| format!("{} isn't a valid sequence number", parts[1]))?;
+    if max < min {
+        Err("invalid sequence number range provided".to_string())
```

Comment:
When I see error messages I like to get told explicitly what is wrong (so I can fix it). Maybe it is obvious in this case but here is my suggestion:

```suggestion
        Err(format!("invalid sequence number range provided: {max} was less than {min}")
```